R/scatterplot_esi.R

```diff
@@ -0,0 +1,54 @@
+#' Generate a Scatterplot of the Earth Similarity Index (ESI)
+#'
+#' Creates a log-log scatterplot of planetary radius versus stellar flux,
+#' colored by the Earth Similarity Index (ESI).
+#'
+#' Dashed lines at (1,1) indicate Earth's reference values for stellar flux and radius.
+#'
+#' @param data A data frame containing exoplanet data. Must include the columns:
+#'   - `pl_insol`: incident stellar flux (in Earth flux units),
+#'   - `pl_rade`: planetary radius (in Earth radii),
+#'   - `esi`: Earth Similarity Index (numeric).
+#' @param data_limits A numeric vector of length 2 specifying the lower and upper bounds.
+#'  Default is \code{c(0.1, 10)}.
+#'
+#' @return A ggplot2 object representing the scatterplot.
+#'
+#' @importFrom ggplot2 ggplot aes geom_point scale_color_viridis_c scale_x_log10
+#'   scale_y_log10 ggtitle theme_bw theme element_blank geom_vline
+#'   geom_hline scale_color_gradient
+#' @importFrom checkmate assert_names assert_data_frame assert_numeric
+#'
+#' @examples
+#' \dontrun{
+#'   data = read.csv("some_esi_dataset.csv")
+#'   scatterplot_esi(data)
+#' }
```

Comment:
issue: Loading non-existing "some_esi_dataset" is pointless as an example. Either use an existing dataset or  define it inline. You have provided an example on how to test in the PR description - why not use it here?

```suggestion
#' @examples
#' data = dplyr::mutate(closest_50_exoplanets, esi = calculate_esi(pl_rade, pl_insol))
#' scatterplot_esi(data)
```